packages/core/data/mm/world/stone_tower_temple.yml

```diff
@@ -56,58 +54,59 @@
   dungeon: ST
   exits:
     "Stone Tower Temple West Garden": "small_keys_st(4) || (small_keys_st(3) && has_mask_goron && (has_explosives || trick_keg_explosives) && can_play(SONG_EMPTINESS))"
-    "Stone Tower Temple Center": "true"
+    "Stone Tower Temple Center": "has_mask_zora || soul(SOUL_DEXIHAND)"
+    "Stone Tower Temple Water Bridge": "(can_goron_bomb_jump && can_use_ice_arrows) || short_hook_anywhere"
   locations:
-    "Stone Tower Temple Center Sun Block Chest": "(has_mask_zora || has(MASK_DEKU) || has_explosives || (has_magic && (has_weapon && has(SPIN_UPGRADE))) || has(SWORD, 3) || has(GREAT_FAIRY_SWORD) || can_use_ice_arrows) && can_use_light_arrows"
+    "Stone Tower Temple Center Sun Block Chest": "short_hook_anywhere || (can_goron_bomb_jump && can_use_ice_arrows && can_use_light_arrows) || (((has(MASK_DEKU) || has_explosives || (has_magic && (has_weapon && has(SPIN_UPGRADE))) || has(SWORD, 3) || has(GREAT_FAIRY_SWORD) || can_use_ice_arrows) && (has_mask_zora || soul(SOUL_DEXIHAND))) && can_use_light_arrows)"
 "Stone Tower Temple Center":
   dungeon: ST
   exits:
-    "Stone Tower Temple Water Room": "has_mask_zora"
+    "Stone Tower Temple Water Room": "has_mask_zora || short_hook_anywhere"
     "Stone Tower Temple Center Ledge": "has_mask_zora"
-    "Stone Tower Temple Water Bridge": "can_goron_bomb_jump && can_use_ice_arrows"
+    "Stone Tower Temple Water Bridge": "(can_goron_bomb_jump && can_use_ice_arrows) || short_hook_anywhere"
   locations:
-    "Stone Tower Temple Center Across Water Chest": "soul(SOUL_DEXIHAND) || has_mask_zora"
+    "Stone Tower Temple Center Across Water Chest": "true"
```

Comment:
Isn't this chest inaccessible without zora or dexihands? I am pretty sure this is on the ledge where the dexihand spawns

Reply:
I changed it so that Center is the ledge with the dexihand, rather than including the water. It's easier that way. You will notice that Center Ledge to Center includes Dexihand or Zora.

Reply:
Not quite certain as to why that change was made. Need more explanation before approving

Reply:
Think about it this way. If you get into that room, what exactly can you do without having access to that ledge? Nothing at all. The Sun Chest requires being able to get onto that ledge, the key chest is on the ledge, getting to eyegore guarantees you can reach that ledge. Being in the water does nothing for the player.

Reply:
And then there's the exit i just fixed. Since if you get there with short hook anywhere, you actually can't do anything without the dexihand.